packages/cli/src/create/add.ts

```diff
@@ -0,0 +1,219 @@
+            []
+          ).map((network) => ({
+            title: network.color(network.display),
+            value: network,
+          })),
+          {
+            title: cyan("Other"),
+            value: {
+              color: cyan,
+              display: "Other",
+              name: "other",
+            } as NetworkDataType,
+          },
+        ],
+      },
+      {
+        type: (network: NetworkDataType | undefined) => {
+          if (network || argNetwork) {
+            return network?.name === "other" || argNetwork === "other"
+              ? "text"
+              : null;
+          }
+          return null;
+        },
+        name: "prompt_dnaUrl",
+        message: reset("Enter a DNA URL:"),
+        validate: (url) => validateDnaUrl(url) || "Provide a valid DNA Url",
+      },
+      {
+        type: argStorage ? null : "select",
+        name: "prompt_storage",
+        message: reset("Select a storage:"),
+        choices: storages.map((storage) => ({
+          title: storage.color(storage.display),
+          value: storage,
+        })),
+      },
+    ],
+    {
+      onCancel: () => {
+        cancelOperation();
+      },
+    },
+  );
+
+  const {
+    prompt_indexerId,
+    prompt_chain,
+    prompt_network,
+    prompt_storage,
+    prompt_dnaUrl,
+  } = result as {
+    prompt_indexerId: string | undefined;
+    prompt_chain: ChainDataType | undefined;
+    prompt_network: NetworkDataType | undefined;
+    prompt_storage: StorageDataType | undefined;
+    prompt_dnaUrl: string | undefined;
+  };
+
+  if (!argIndexerId && !prompt_indexerId) {
+    throw new Error(red("✖") + " Indexer ID is required");
+  }
+
+  if (!argChain && !prompt_chain) {
+    throw new Error(red("✖") + " Chain is required");
+  }
+
+  if (!argNetwork && !prompt_network) {
+    throw new Error(red("✖") + " Network is required");
+  }
+
+  const indexerFileId = argIndexerId! ?? prompt_indexerId!;
+
+  const options: IndexerOptions = {
+    cwd: process.cwd(),
+    indexerFileId,
+    indexerId: convertKebabToCamelCase(indexerFileId),
+    chain: (argChain as Chain) ?? prompt_chain?.name!,
+    network: (argNetwork as Network) ?? prompt_network?.name!,
+    storage: (argStorage as Storage) ?? prompt_storage?.name!,
+    dnaUrl: argDnaUrl ?? prompt_dnaUrl,
+    language,
+  };
+
+  updateApibaraConfigFile(options);
+
+  consola.success(
+    `Updated ${cyan("apibara.config." + (language === "typescript" ? "ts" : "js"))}`,
+  );
+
+  updatePackageJson(options);
+
+  consola.success(`Updated ${cyan("package.json")}`);
+
+  await createIndexerFile(options);
+
+  consola.success(
+    `Created ${cyan(`${indexerFileId}.indexer.${language === "typescript" ? "ts" : "js"}`)}`,
+  );
+
+  await createStorageRelatedFiles(options);
+
+  console.log();
+
+  consola.info(
+    `Before running the indexer, run ${cyan("npm run install")}${
+      language === "typescript" ? " & " + cyan("npm run prepare") : ""
+    }`,
+  );
```

Comment:
We should be detecting the package manager used by the user and propagate it to all helpers through `IndexerOptions`